geoh5py/groups/property_group.py

```diff
@@ -355,41 +352,52 @@ def verify_data(self, data: Data | UUID | str) -> UUID:
 
         :return: The uuid of the data.
         """
-        data = str2uuid(data)
+        data = self.reference_to_data(data)
 
```

Comment:
```suggestion
self._association = data.association if self.association is None

```

Reply:
Rather be explicit about checking for association.